compiler/rustc_driver/src/pretty.rs

```diff
@@ -375,13 +373,14 @@ fn get_source(input: &Input, sess: &Session) -> (String, FileName) {
     (src, src_name)
 }
 
-fn write_output(out: Vec<u8>, ofile: Option<&Path>) {
+fn write_or_print(out: &str, ofile: Option<&Path>) {
```

Comment:
```suggestion
fn write_or_print(out: &[u8], ofile: Option<&Path>) {
```
This would still avoid allocations, but remove the need for a `str::from_utf8` call, which needs to validate if the input is valid utf-8.

Reply:
The prior code had a String::from_utf8 call inside. If out is a &[u8] I can change it to `std::io::stdout().write_all(out.as_ref()).unwrap(),`.

Which do you prefer?

Reply:
Missed that. Keeping `&str` is fine I think.